### PR TITLE
Ehealth 1233 Declaration Verification

### PR DESCRIPTION
**Pull request by ybod**
Additional verification logic for Create Declaration Request

https://github.com/edenlabllc/ehealth.api/issues/1233

### CODE REVIEW

File: lib/ehealth/declaration_request/api/validate_person.ex [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+       "PHONE_TYPE",
+       "AUTHENTICATION_METHOD",
+       "DOCUMENT_RELATIONSHIP_TYPE"])
+
+    confidant_persons = Map.get(person, "confidant_person")
+
+    with  %{"DOCUMENT_TYPE" => doc_types} = dictionaries,
+          {:ok, _} <- JsonObjects.array_unique_by_key?(Map.get(person, "documents"), "type", Map.keys(doc_types)),
+          %{"PHONE_TYPE" => phone_types} = dictionaries,
+          {:ok, _} <- JsonObjects.array_unique_by_key?(Map.get(person, "phones"), "type", Map.keys(phone_types)),
```
Comment (Troush) [inline lib/ehealth/declaration_request/api/validate_person.ex]: Why you don't catch an error message that you pass from from `JsonObject`?
So maybe you detach error message logic from `JsonObject` and only get a status from it and then as you already do implement error message near the specific check implementation

File: lib/ehealth/declaration_request/api/validate_person.ex [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  defp validate_confidant_persons([], _), do:   :ok
+
+  defp validate_confidant_persons(confidant_persons, dict_keys) do
+    with  :ok <- JsonObjects.array_unique_by_key?(confidant_persons, "relation_type", ["PRIMARY", "SECONDARY"]),
+          :ok <- JsonObjects.array_contains_item?(confidant_persons, "relation_type", "PRIMARY"),
+          :ok <- validate_every_confidant_person(confidant_persons, dict_keys),
+    do:   :ok
+  end
+
+  defp validate_every_confidant_person([], _), do:
```
Comment (Troush) [inline lib/ehealth/declaration_request/api/validate_person.ex]: It's better to group function headers without blank line and place zero termination clauses under main logic function

File: lib/ehealth/dictionaries/dictionaries.ex [hunk 3]
```diff
@@ -89,4 +89,17 @@ defmodule EHealth.Dictionaries do
...
+
+  def get_dictionaries(dictionary_list) do
+    q = from d in Dictionary,
+        where: d.name in ^dictionary_list and d.is_active,
+        select: {d.name, d.values}
+
+    Map.new(Repo.all(q))
+  end
+
+  def get_dictionaries_keys(dictionary_list) do
```
Comment (Troush) [inline lib/ehealth/dictionaries/dictionaries.ex]: Pipes must start with bare value

File: test/support/data_case.ex [hunk 4]
```diff
@@ -0,0 +1,58 @@
+defmodule EHealth.Web.DataCase do
```
Comment (Troush) [inline test/support/data_case.ex]: Are we really need this file? Whats wrong with ConnCase?

Comment (ybod) [reply test/support/data_case.ex]: removed

**Pull request merged**